packages/react-instantsearch-dom/src/core/__tests__/createInstantSearchServer.js

```diff
@@ -503,6 +524,78 @@ describe('findResultsState', () => {
         expect(data.rawResults[0].params).toMatchInlineSnapshot(`"lol=lol"`);
       });
     });
+
+    it('searches twice (cached) with dynamic widgets', async () => {
+      const RefinementList = connectRefinementList(() => null);
+      const App = (props) => (
+        <InstantSearch {...props}>
+          <DynamicWidgets fallbackComponent={RefinementList} />
+        </InstantSearch>
+      );
+
+      const props = {
+        searchClient: createSearchClient(),
+        indexName: 'abc',
+        searchState: {
+          query: 'iPhone',
+        },
+      };
+
+      await findResultsState(App, props);
+
+      expect(props.searchClient.search).toHaveBeenCalledTimes(2);
+      // both calls are the same, so they're cached
+      expect(props.searchClient.search.mock.calls[0][0]).toEqual(
+        props.searchClient.search.mock.calls[1][0]
+      );
+    });
+
+    it('searches twice with dynamic widgets and a refinement', async () => {
+      const RefinementList = connectRefinementList(() => null);
+      const App = (props) => (
+        <InstantSearch {...props}>
+          <DynamicWidgets fallbackComponent={RefinementList} />
+        </InstantSearch>
+      );
+
+      const props = {
+        searchClient: createSearchClient(),
+        indexName: 'instant_search',
+        searchState: {
+          query: 'iPhone',
+          refinementList: { categories: ['refined!'] },
+        },
+      };
+
+      await findResultsState(App, props);
+
+      expect(props.searchClient.search).toHaveBeenCalledTimes(2);
+
+      // first query doesn't have the fallback widget mounted yet
+      expect(props.searchClient.search.mock.calls[0][0][0]).toEqual({
+        indexName: 'instant_search',
+        params: {
+          facets: ['*'],
+          highlightPostTag: '</ais-highlight-0000000000>',
+          highlightPreTag: '<ais-highlight-0000000000>',
+          maxValuesPerFacet: 20,
+          tagFilters: '',
+        },
+      });
+
+      // second query does have the fallback widget mounted, and thus also the refinement
+      expect(props.searchClient.search.mock.calls[1][0][0]).toEqual({
+        indexName: 'instant_search',
+        params: {
+          facetFilters: [['categories:refined!']],
```

Comment:
according to the test, facetFilters isn't present in this case, but it is in multi-index context. When I'm trying it out manually in the Next app, it seems that the right parameters are used.

Reply:
so in the test the refinement list doesn't mount, but it does mount when it's nested in an index. as far as I can tell the results are passed correctly though, so I don't yet see the difference between the two cases

Reply:
solved in 77ce0f1d